torch/_inductor/lowering.py

```diff
@@ -6717,7 +6717,7 @@ def triton_kernel_wrap_(
     return {key: val for key, val in kwargs.items() if isinstance(val, TensorBox)}
 
 
-@register_lowering(torch.ops.higher_order.cond)
+@register_lowering(torch.ops.higher_order.cond, type_promotion_kind=None)
```

Comment:
Do you have a test?

Reply:
It's the same as other hops and we can't test it because type promotion doesn't handle tuple inputs (so it accidentally works before).